frontend/src/concepts/pipelines/apiHooks/useLatestPipelineVersion.ts

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import { PipelineVersionKFv2 } from '~/concepts/pipelines/kfTypes';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import useFetchState, {
+  FetchState,
+  FetchStateCallbackPromise,
+  NotReadyError,
+} from '~/utilities/useFetchState';
+
+export const useLatestPipelineVersion = (
+  pipelineId: string | undefined,
+): FetchState<PipelineVersionKFv2 | null> => {
+  const { api } = usePipelinesAPI();
+
+  const getLatestVersion = React.useCallback<
+    FetchStateCallbackPromise<PipelineVersionKFv2 | null>
+  >(async () => {
+    if (!pipelineId) {
+      return Promise.reject(new NotReadyError('No pipeline id'));
+    }
+
+    const response = await api.listPipelineVersions({}, pipelineId, {
+      sortField: 'created_at',
+      sortDirection: 'desc',
+    });
```

Comment:
Curious, can we reduce this to a page size of 1? This will default to 10 and we don't care about 9 of those.

Reply:
Yes, I can do that.